Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -384,6 +384,17 @@
         return;
     }
 
+    if (!state->hasReceivedText && session.compositionType == WritingTools::Session::CompositionType::SmartReply) {
+        // UIKit inserts a space prior to `willBegin` and before `didReceiveText`, so the initial session range
+        // becomes invalid and must be re-computed.
+        //
+        // FIXME: Remove this special-case logic if/when UIKit no longer injects a space.
+        if (auto selectedRange = document->selection().selection().firstRange())
+            state->reappliedCommands.last()->setEndingContextRange(*selectedRange);
+    }
+
+    state->hasReceivedText = true;
```

Comment:
Does this need to get reset in a 'restart' case? Does that kind of case exist for smart replies?

Reply:
no such case exists for smart replies